src/Talis/Persona/Client/Base.php

```diff
@@ -92,4 +102,54 @@ protected function checkConfig($config){
             throw new \InvalidArgumentException("Config provided does not contain values for: " . implode(",", $missingProperties));
         }
     }
+
+    /**
+     * @return Logger|\Psr\Log\LoggerInterface
+     */
+    protected function getLogger()
+    {
+        if ($this->logger==null)
+        {
+            $this->logger = new Logger(self::LOGGER_NAME);
+        }
+        return $this->logger;
+    }
+
+    /**
+     * Perform the request according to the $curlOptions
+     * @param $curlOptions
+     * @param $expectResponse set true if you expect a JSON response with a 200, otherwise expect a 204 no content
+     * @return array
+     * @throws \Exception if response not 200 and valid JSON
+     */
+    protected function performRequest($curlOptions,$expectResponse=true)
+    {
+        $curl = curl_init();
+        curl_setopt_array($curl, $curlOptions);
+
+        $response = curl_exec($curl);
+        $headers = curl_getinfo($curl);
+        curl_close($curl);
+
+        $expectedResponseCode = ($expectResponse) ? 200 : 204;
+        if (isset($headers['http_code']) && $headers['http_code'] === $expectedResponseCode)
+        {
+            if ($expectResponse) // expect JSON!
+            {
+                $json = json_decode($response,true);
+                if (empty($json))
+                {
+                    $this->getLogger()->error("Could not parse response {$response} as JSON");
+                    throw new \Exception("Could not parse response as JSON");
+                }
+                return $json;
+            }
+        }
+        else
+        {
+            $this->getLogger()->error("Did not retrieve successful response code: {$headers['http_code']}");
```

Comment:
Given that one of the conditions that could of gotten us here is that 'http_code' isn't set...